roles/cifmw_cephadm/tasks/rgw_cert.yml

```diff
@@ -0,0 +1,119 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if cifmw_cephadm_certificate path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_certificate }}"
+  register: cert_stat_result
+  when: cifmw_cephadm_certificate | length > 0
+
+- name: Check if cifmw_cephadm_key path exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_cephadm_key }}"
+  register: key_stat_result
+  when: cifmw_cephadm_key | length > 0
+
+- name: Create certificate if paths do not exist on target host
+  when: (not cert_stat_result.stat.exists | bool) or
+        (not key_stat_result.stat.exists | bool)
+  block:
+    - name: Ensure we have cifmw_openshift_kubeconfig
+      delegate_to: localhost
+      ansible.builtin.stat:
+        path: "{{ cifmw_openshift_kubeconfig }}"
+      register: cifmw_openshift_kubeconfig_stat_result
+
+    - name: Fail if kube config file is missing
+      delegate_to: localhost
+      fail:
+        msg: >-
+          Missing kube config file ({{ cifmw_openshift_kubeconfig }})
+          Canot create RGW certificate without access to root CA in k8s
+      when:
+        - not cifmw_openshift_kubeconfig_stat_result.stat.exists | bool
+
+    - name: Create Ceph certificate manifest directory
+      delegate_to: localhost
+      ansible.builtin.file:
+        path: "{{ cifmw_cephadm_manifests_dir }}"
+        state: directory
+
+    - name: Put the certificate manifest file(s) in place
+      delegate_to: localhost
+      ansible.builtin.template:
+        src: "templates/{{ item }}.yml.j2"
+        dest: "{{ cifmw_cephadm_manifests_dir }}/{{ item }}.yml"
+        mode: "0644"
+        force: true
+      loop:
+        - cert_ceph_rgw
+
+    - name: Apply k8s manifest file of kind Certficate for Ceph RGW
+      delegate_to: localhost
+      kubernetes.core.k8s:
+        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+        api_key: "{{ cifmw_openshift_token | default(omit) }}"
+        context: "{{ cifmw_openshift_context | default(omit) }}"
+        src: "{{ cifmw_cephadm_manifests_dir }}/cert_ceph_rgw.yml"
```

Comment:
should  `cert_ceph_rgw` be part of a loop like the previous task? Also, I'm wondering if we can use `cifmw_cephadm_cert_name` here

Reply:
ack

```diff
-        src: "{{ cifmw_cephadm_manifests_dir }}/cert_ceph_rgw.yml"
+        src: "{{ cifmw_cephadm_manifests_dir }}/{{ cifmw_cephadm_cert_name }}.yml"
```